File: TerminBA/TerminBA.Models/Model/PostResponse.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TerminBA.Models.Model
{
    public class PostResponse
    {
        public int Id { get; set; }

        public string? SkillLevel { get; set; }

        public string? Text { get; set; }

        public int ReservationId { get; set; }

        public ReservationResponse? Reservation { get; set; }
    }
}

File: TerminBA/TerminBA.UI/terminba_mobile/lib/model/post_response.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:terminba_mobile/model/reservation_response.dart';

part 'post_response.g.dart';

@JsonSerializable(explicitToJson: true)
class PostResponse {
  final int id;
  final String? skillLevel;
  final String? text;
  final int reservationId;
  final ReservationResponse? reservation;
  final int numberOfPlayersWanted;
  final int numberOfPlayersFound;
  final String postState;

  const PostResponse({
    required this.id,
    this.skillLevel,
    this.text,
    required this.reservationId,
    this.reservation,
    required this.numberOfPlayersWanted,
    required this.numberOfPlayersFound,
    required this.postState,
  });

  bool get isActive => postState == 'PlayerSearchPostState';
  bool get isClosed =>
      postState == 'ClosedPostState' || postState == 'PlayerFoundPostState';

  factory PostResponse.fromJson(Map<String, dynamic> json) =>
      _$PostResponseFromJson(json);

  Map<String, dynamic> toJson() => _$PostResponseToJson(this);
}
